include/boost/compute/algorithm/detail/find_extrema.hpp

```diff
@@ -32,11 +33,25 @@ inline InputIterator find_extrema(InputIterator first,
         return first;
     }
 
+    const device &device = queue.get_device();
+
     // use serial method for small inputs
-    if(count < 64){
+    // and when device is a CPU
+    if(count < 64 || (device.type() & device::cpu)){
         return serial_find_extrema(first, last, sign, queue);
     }
 
+    // find_extrema_reduce() is used only if requirements are met
+    size_t preferred_work_group_size = 256;
```

Comment:
OK. In that case I think I will add ```work_group_size```  and ```work_items_per_compute_unit``` parameters, because they both influence performance of the ```find_extrema_reduce```.